File: go/luhn/luhn.go
package luhn

import "strings"

// Valid checks a number for Luhn validity.
func Valid(num string) bool {
	num = strings.Replace(num, " ", "", -1)
	if len(num) <= 1 {
		return false
	}
	sum := 0
	for i := range num {
		newNum := int(num[len(num)-1-i] - '0')
		if newNum < 0 || newNum > 9 {
			return false
		}
		if i%2 == 1 {
			newNum = newNum * 2
			if newNum > 9 {
				newNum -= 9
			}
		}
		sum += newNum
	}

	return sum%10 == 0
}

File: javascript/luhn/luhn.js
export class Luhn {
  constructor(num) {
    const sum = num.split('')
      .filter(n => n != ' ')
      .map(n => Number(n))
      .reverse()
      .reduce((accum, d, i) => {
        if (i % 2 == 1) {
          const prod = d * 2;
          if (prod> 9) {
            accum += (prod - 9);
          } else {
            accum += (prod);
          }
        } else {
          accum += d;
        }
        return accum;
      }, 0);
    this.valid = sum != 0 && sum % 10 == 0;
  }
}